engine/src/mac-qt-recorder.mm

```diff
@@ -808,6 +796,31 @@ static bool available_compression_ids(void *context, unsigned int p_id, const ch
     return true;
 }
 
+struct FormatTable
+{
+    const char *label;
+    OSType value;
+};
+
+static FormatTable record_formats[] =
+{
+    { "aiff", kQTFileTypeAIFF },
+    { "wave", kQTFileTypeWave },
+    { "ulaw", kQTFileTypeMuLaw },
+    { "movie", kQTFileTypeMovie },
+};
+
+bool MCQTSoundRecorder::ListFormats(MCPlatformSoundRecorderListFormatsCallback p_callback, void *context)
+{
+    for (int i = 0; i < sizeof(record_formats) / sizeof(record_formats[0]); i++)
+    {
+        if (!p_callback(context, record_formats[i].value, MCSTR(record_formats[i].label)))
+            return false;
+    }
```

Comment:
This might be a good opportunity to use `<algorithm>`.  For example:

```c++
return std::all_of(std::begin(record_formats), std::end(record_formats),
    [&](const FormatTable& p_fmt) {
        return p_callback(context, p_fmt.value, MCSTR(p_fmt.label));
    });
```